bindings/cs/rl.net.native/rl.net.factory_context.cc

```diff
@@ -0,0 +1,74 @@
+#include "rl.net.factory_context.h"
+#include "error_callback_fn.h"
+
+using namespace reinforcement_learning;
+
+API factory_context_t* CreateFactoryContext()
+{
+  auto context = new factory_context_t;
+  
+  context->data_transport_factory = &data_transport_factory;
+  context->model_factory = &model_factory;
+  context->sender_factory = &sender_factory;
+  context->trace_logger_factory = &trace_logger_factory;
+  context->time_provider_factory = &time_provider_factory;
+
+  return context;
+}
+
+void cleanup_trace_logger_factory(trace_logger_factory_t* trace_logger_factory)
+{
+  if (trace_logger_factory != nullptr &&
+      trace_logger_factory != &reinforcement_learning::trace_logger_factory)
+  {
+    // We overrode the built-in trace_logger_factory
+    delete trace_logger_factory;
+  }
+}
+
+void cleanup_sender_factory(sender_factory_t* sender_factory)
+{
+  if (sender_factory != nullptr && 
+      sender_factory != &reinforcement_learning::sender_factory)
+  {
+    // We overrode the built-in sender_factory
+    delete sender_factory;
+  }
+}
+
+API void DeleteFactoryContext(factory_context_t* context)
+{
+  cleanup_trace_logger_factory(context->trace_logger_factory);
+  cleanup_sender_factory(context->sender_factory);
+
+  // TODO: Once we project the others, we will need to add them to cleanup.
```

Comment:
should this TODO be documented as a issue/ticket?

Reply:
This is there to ensure that anyone working on projecting out the rest of the extensibility points will make sure they add them to the cleanup function, since there is no mechanism to actually change them away from the global factories right now.

The alternative is to add the code right now, so it does not lead to an issue in the future, but at the cost of superfluous code (in practice).